### PR TITLE
Lookup Service

### PR DESCRIPTION
**Pull request by aswin90**
Service to return address when queried with NEU locations

### CODE REVIEW

File: src/main/java/com/neu/wham/controllers/LookupController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+@Controller
+public class LookupController {
+	
+	
+	
+	@Autowired
+	private LookupService lookupService;
+	
+	@RequestMapping(value = "/lookup/{location}", method = RequestMethod.GET)
```
Comment (millayr) [inline src/main/java/com/neu/wham/controllers/LookupController.java:29]: What is the location?  Can you add some comments describing this?  Is it a name or an address?

File: src/main/java/com/neu/wham/services/LookupServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
+package com.neu.wham.services;
+
+import java.util.*;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class LookupServiceImpl implements LookupService {
+	
+	Map<String, String> abbvSource;
```
Comment (millayr) [inline src/main/java/com/neu/wham/services/LookupServiceImpl.java:10]: Do we have all of the buildings and addresses we need for campus?

File: src/main/java/com/neu/wham/services/LookupServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	@Override
+	public String lookup(String location) {
+		// TODO Auto-generated method stub
+		if(abbvSource.containsKey(location)){
+			return (abbvSource.get(location) + ", Boston, MA");
+		}
+		
+		return "error";
```
Comment (millayr) [inline src/main/java/com/neu/wham/services/LookupServiceImpl.java:124]: Is returning "error" better than throwing an exception?  I'm not arguing either way, just curious.

**Comment by millayr** [comment]
General comment - I'm curious how this fits in with the larger project.  Does the frontend team need this and do they know what the key values are?

File: src/main/java/com/neu/wham/services/LookupServiceImpl.java (hunk 2 above)
Comment (aswin90) [inline src/main/java/com/neu/wham/services/LookupServiceImpl.java:10]: Contains all the locations from - https://www.northeastern.edu/campusmap/map/qad4.html and select important ones from https://www.northeastern.edu/campusmap/map/index.html.

File: src/main/java/com/neu/wham/controllers/LookupController.java (hunk 1 above)
Comment (aswin90) [inline src/main/java/com/neu/wham/controllers/LookupController.java:29]: Sure, will do!

**Comment by aswin90** [comment]
Yes, this was a requirement that came out from previous meeting with the
client -
https://cs5500.ccs.neu.edu/confluence/pages/viewpage.action?pageId=6165226.
The client wants to enter WVH and create an event at the specified
location. So we are coming up with this service, so that we know where to
display WVH on maps :)

File: src/main/java/com/neu/wham/services/LookupServiceImpl.java (hunk 3 above)
Comment (aswin90) [inline src/main/java/com/neu/wham/services/LookupServiceImpl.java:124]: In the discussion with the front end team, I said, I will return "error". Agreed to throwing an exception, probably, 404 is the best approach. We can go either way, based on what the front end team expects!

**Pull request merged**